presto-main/src/test/java/io/prestosql/sql/query/TestSubqueries.java

```diff
@@ -250,6 +250,19 @@ public void testCorrelatedSubqueriesWithLimit()
                         "   WHERE outer_relation.id = inner_relation.id ORDER BY value LIMIT 2) " +
                         "ON TRUE",
                 "VALUES (1, null), (2, null), (3, 'a'), (3, 'b'), (null, null)");
+        // do not fail to decorrelate the innermost subquery which actually is not correlated
+        assertions.assertQuery(
```

Comment:
maybe wrap it with separate method like with the test below. Maybe a group of tests with uncorrelated subquery wrapped in correlated subquery?